src/coreclr/jit/emitxarch.h

```diff
@@ -170,6 +421,52 @@ void SetUseVEXEncoding(bool value)
     useVEXEncodings = value;
 }
 
+bool useEVEXEncodings;
+bool UseEVEXEncoding() const
+{
+    return useEVEXEncodings;
+}
+void SetUseEVEXEncoding(bool value)
+{
+    useEVEXEncodings = value;
+}
+
+// 4-byte EVEX prefix starts with byte 0x62
+#define EVEX_PREFIX_MASK 0xFF00000000000000ULL
+#define EVEX_PREFIX_CODE 0x6200000000000000ULL
+
+bool TakesEvexPrefix(instruction ins) const;
+
+// Returns true if the instruction encoding already contains EVEX prefix
+bool hasEvexPrefix(code_t code)
+{
+    return (code & EVEX_PREFIX_MASK) == EVEX_PREFIX_CODE;
+}
+code_t AddEvexPrefix(instruction ins, code_t code, emitAttr attr);
+code_t AddSIMDPrefixIfNeeded(instruction ins, code_t code, emitAttr size)
+{
+    if (TakesEvexPrefix(ins))
+    {
+        code = AddEvexPrefix(ins, code, size);
+    }
+    else if (TakesVexPrefix(ins))
+    {
+        code = AddVexPrefix(ins, code, size);
+    }
+    return code;
+}
+
+bool hasSIMDPrefix(code_t code)
```

Comment:
I wonder if this method will be "expected" to also report legacy SIMD prefixes.

Those aren't particularly easy to detect, however.

Reply:
Maybe `hasVexOrEvexPrefix(ins)` would be a better name given how its used later on.

Reply:
Changed.